### PR TITLE
ENT-11338: Adjusted cfbs packaging on rhel-8 to use platform-python

### PR DESCRIPTION
**Pull request by craigcomstock**
Ticket: ENT-11338
Changelog: title

### CODE REVIEW

**Comment by craigcomstock** [comment]
[Build Status](https://ci.cfengine.com/job/pr-pipeline/10464/)

**Comment by craigcomstock** [comment]
got a build from latest changes, testing
[Build Status](https://ci.cfengine.com/job/pr-pipeline/10471/)

**Comment by craigcomstock** [comment]
looks good in the package

```
[root@rhel8 ~]# /var/cfengine/bin/cfbs  -V
cfbs 4.0.8
[root@rhel8 ~]# head -1 /var/cfengine/bin/cfbs 
#! /usr/libexec/platform-python -s
```

**Comment by craigcomstock** [comment]
checking rhel-9 hub build, shouldn't need changes to build host at all since that's rhel-8 only. :)
[Build Status](https://ci.cfengine.com/job/pr-pipeline/10478/)

**Review by nickanderson — DISMISSED** [review]

**Comment by craigcomstock** [comment]
[Build Status](https://ci.cfengine.com/job/pr-pipeline/10479/)

**Comment by craigcomstock** [comment]
[Build Status](https://ci.cfengine.com/job/pr-pipeline/10480/)

**Comment by craigcomstock** [comment]
double check rhel-8 [Build Status](https://ci.cfengine.com/job/pr-pipeline/10481/)

**Review by nickanderson — APPROVED** [review]

File: ci/cfengine-build-host-setup.cf [hunk 1]
```diff
@@ -107,6 +107,10 @@ bundle agent cfengine_build_host_setup
 
     (redhat_8|centos_8).(yum_dnf_conf_ok)::
       "python2-psycopg2" comment => "This will bring in python2";
+      "python3-rpm-macros" -> { "cfbs shebang", "ENT-11338" }
```
Comment (vpodzime) [inline ci/cfengine-build-host-setup.cf:110]: What do RPM macros have to do with the shebang in `cfbs`?

**Review by vpodzime — COMMENTED** [review]
The commit is quite confusing to me.

File: ci/cfengine-build-host-setup.cf (hunk 1 above)
Comment (craigcomstock) [reply ci/cfengine-build-host-setup.cf]: I adjusted the policy and the spec file to make this more clear. python3-rpm-macros provides py3_shebang_fix macro which munges the /usr/bin/python3 shebang in /var/cfengine/bin/cfbs into what will work on rhel-8 which is /usr/libexec/platform-python provided by the platform-python package which is included by default there.

This switched back again in rhel-9 AFAICT where dnf depends on python39 and platform-python package is gone. (double checking this last statement now and will update this comment)

**Comment by craigcomstock** [comment]
@cf-bottom jenkins please

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/10482/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/10482/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-10482/

File: ci/cfengine-build-host-setup.cf (hunk 1 above)
Comment (craigcomstock) [reply ci/cfengine-build-host-setup.cf]: yep, not there anymore

```
[root@rhel9 vagrant]# yum info platform-python
Updating Subscription Management repositories.
Last metadata expiration check: 0:46:15 ago on Fri 16 Feb 2024 02:48:00 PM UTC.
Error: No matching Packages to list
```

**Comment by craigcomstock** [comment]
rhel-8-hub testing-pr failure is expected as this PR also includes the bits to change the build hosts spawned by jenkins to include needed packages. Should be fine. Todo, streamline how CI works by spawning instances from this repo instead of using jenkins cloud config: https://northerntech.atlassian.net/browse/ENT-5964

**Comment by craigcomstock** [comment]
CI failures are flakes. I think this is "good to go".

**Comment by craigcomstock** [comment]
last commit of two whitespace lines removed and squashed. still ready to go. :)

**Review by vpodzime — APPROVED** [review]
Makes sense now. Thanks!

**Comment by craigcomstock** [comment]
3.18.x doesn't have cfbs/build so cherry picked only to 3.21.x
https://github.com/cfengine/buildscripts/pull/1399

**Pull request merged**